src/internal/transport/AbstractRequestProcessor.ts

```diff
@@ -244,14 +244,23 @@ export abstract class AbstractRequestProcessor implements RequestProcessor {
                           callback: (results: Promise<HandlerResult[]>) => void) {
 
         const finalize = (results: HandlerResult[]) => {
-            this.sendEventStatus(results.some(r => r.code !== 0) ? false : true, ef, event, ctx)
+            let noncircularResults = results;
+            try {
+                JSON.stringify(noncircularResults);
+            } catch (err) {
+                logger.error("Circular object returned from event handler: %s", stringify(results));
+                noncircularResults = results.map(r => ({ code: r.code, message: stringify(r.message) }));
+                logger.error("Substituting for circular object: %j", noncircularResults);
```

Comment:
Is logging statement needed since we log it either way below?

Reply:
I wanted to be explicit about the substitution that was made.

Reply:
Doesn't the first log statement make it clear that a substitution will be made?